src/services/DiscourseService.ts

```diff
@@ -0,0 +1,72 @@
+import logger from 'decentraland-gatsby/dist/entities/Development/logger'
+import { Avatar } from 'decentraland-gatsby/dist/utils/api/Catalyst'
+
+import { Discourse, DiscoursePost } from '../clients/Discourse'
+import { DISCOURSE_CATEGORY } from '../entities/Discourse/utils'
+import * as templates from '../entities/Proposal/templates'
+import { forumUrl, proposalUrl } from '../entities/Proposal/utils'
+import { inBackground } from '../helpers'
+
+import { ProposalInCreation } from './ProposalService'
+import { SnapshotService } from './SnapshotService'
+
+export class DiscourseService {
+  static async createProposal(
+    data: ProposalInCreation,
+    proposalId: string,
+    profile: Avatar | null,
+    snapshotUrl: string,
+    snapshotId: string
+  ) {
+    try {
+      const discoursePost = await this.getDiscoursePost(data, profile, proposalId, snapshotUrl, snapshotId)
+      const discourseProposal = await Discourse.get().createPost(discoursePost)
+      this.logProposalCreation(discourseProposal)
+      return discourseProposal
+    } catch (error: any) {
+      SnapshotService.dropSnapshotProposal(snapshotId)
```

Comment:
To discuss later: maybe the create proposal function can receive an `onError` function so this service doesn't know about this and move this logic to the proposal creation flow.